backend/embeddings/embeddings.go

```diff
@@ -117,3 +120,43 @@ func (c *OpenAIClient) GetEmbeddings(ctx context.Context, errors []*model.ErrorO
 
 	return lo.Values(results), nil
 }
+
+func (c *OpenAIClient) GetErrorTagEmbedding(ctx context.Context, title string, description string) (*model.ErrorTag, error) {
+	resp, err := c.client.CreateEmbeddings(
+		context.Background(),
+		openai.EmbeddingRequest{
+			Input: []string{title + " " + description},
+			Model: openai.AdaEmbeddingV2,
+			User:  "highlight-io",
+		},
+	)
+	var embedding = resp.Data[0].Embedding
+	errorTag := &model.ErrorTag{
+		Title:       title,
+		Description: description,
+		Embedding:   embedding,
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	return errorTag, nil
+
+}
+
+func (c *OpenAIClient) GetStringEmbedding(ctx context.Context, text string) ([]float32, error) {
+	resp, err := c.client.CreateEmbeddings(
+		context.Background(),
+		openai.EmbeddingRequest{
+			Input: []string{text},
+			Model: openai.AdaEmbeddingV2,
+			User:  "highlight-io",
+		},
+	)
```

Comment:
could we deduplicate this code? can `GetErrorTagEmbedding` just call `GetStringEmbedding`?